### PR TITLE
add headers: compare, cfloat, cinttypes, climits, cstdarg, modify headers: utility, algorithm, new, memory, cstddef

### PR DESCRIPTION
**Pull request by ilobilo**
added headers: cfloat, cinttypes, climits, cstdarg, compare

* compare
  - partial_ordering
  - weak_ordering
  - strong_ordering
  - common_comparison_category/_t
  - compare_three_way_result/_t
  - compare_three_way
  - is_eq
  - is_neq
  - is_lt
  - is_lteq
  - is_gt
  - is_gteq

* cstddef
  - std::ptrdiff_t

* cinttypes
  - std::imaxdiv_t

* new
  - align_val_t
  - destroying_delete/_t

* algorithm
  - constexpr min/max

* utility
  - piecewise_construct/_t
  - in_place/_t
  - in_place_type/_t
  - in_place_index/_t

### CODE REVIEW

File: stage2/include/memory [hunk 1]
```diff
@@ -14,11 +16,37 @@ T *addressof(T &r) noexcept {
...
+	{
+		::operator delete(ptr);
+	}
+
+	friend constexpr bool operator==(const allocator &, const allocator &) noexcept
+	{
+		return true;
+	}
+};
+
```
Comment (ArsenArsen) [inline stage2/include/memory:49]: This is not even in P1642, are you sure you want to add it?

File: stage2/include/climits [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#ifndef _CXXSHIM_CLIMITS
+#define _CXXSHIM_CLIMITS
+
+#include <limits.h>
+
```
Comment (ArsenArsen) [inline stage2/include/climits:5]: these appear to miss `std` aliases, might want to add them (similarly in other headers)

File: stage2/include/cstddef [hunk 3]
```diff
@@ -8,7 +8,8 @@ namespace std {
 enum class [[gnu::may_alias]] byte : unsigned char { };
 
 using nullptr_t = decltype(nullptr);
-using size_t = ::size_t;
```
Comment (ArsenArsen) [inline stage2/include/cstddef:11]: unrelated change

File: stage2/include/memory [hunk 4]
```diff
@@ -14,11 +16,37 @@ T *addressof(T &r) noexcept {
         // __has_builtin macro
 	return __buitin_addressof(r);
 #else
-	return reinterpret_cast<T *>(
-      const_cast<char *>(&reinterpret_cast<const volatile char &>(r)));
+	return reinterpret_cast<T *>(const_cast<char *>(&reinterpret_cast<const volatile char &>(r)));
```
Comment (ArsenArsen) [inline stage2/include/memory:19]: unrelated change

File: stage2/include/utility [hunk 5]
```diff
@@ -63,6 +63,34 @@ using make_index_sequence = make_integer_sequence<size_t, N>;
...
+
+template<size_t I>
+struct in_place_index_t
+{
+    explicit in_place_index_t() = default;
+};
+
+template<size_t I>
+inline constexpr in_place_index_t<I> in_place_index { };
+
```
Comment (ArsenArsen) [inline stage2/include/utility:93]: this commit summary line is too long (d036ab00fefd31bc1d18f92c3fd8a2b78f9aea3e)

**Review by ArsenArsen — COMMENTED** [review]
where does this code come from? the code style seems all over the place

EDIT: unsure how much we want to grow cxxshim anyway, especially outside of the bounds of P1642, considering it'd be effectively obsoleted by GCC13

File: stage2/include/compare [hunk 6]
```diff
@@ -0,0 +1,344 @@
+#ifndef _CXXSHIM_COMPARE
```
Comment (ArsenArsen) [inline stage2/include/compare:1]: also this commit message (dda47138d6504c6a471c53621e44005b285c98fc) has an excessively long line

File: stage2/include/climits (hunk 2 above)
Comment (ilobilo) [reply stage2/include/climits:5]: climits and cfloat only have macros in them
cinttypes has std::imaxdiv_t which I added
